streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImpl.java

```diff
@@ -224,12 +225,25 @@
                         + " retention=[" + retentionPeriod + "]");
             }
 
-            supplier = Stores.persistentTimestampedWindowStore(
+            // get from default store implementation
+            final StoreImplementation storeImplementation = materialized.storeImplementation();
```

Comment:
This is a meta comment: I feel the introduction of the `StoreImplementation` is a bit overkill here. Could the same be achieved with just an `enum` type say "StoreImplTypes" kept inside the Materialize object, and inside `Stores` the added creator would be:

```
Stores#keyValueBytesStoreSupplier(storeImplTypes, storeName, ...)
Stores#windowBytesStoreSupplier(storeImplTypes, storeName, ...)
Stores#sessionBytesStoreSupplier(storeImplTypes, storeName, ...)
```

Then inside the impl classes like here we would could directly do:

```
if ((supplier = materialized.storeSupplier) == null) {
    supplier = Stores.windowBytesStoreSupplier(materialized.storeImplType())
}
```